### ISSUE TITLE
Make _raw data in Kubernetes Secrets optional

### ISSUE THREAD

**Issue by rgarcia89** [hashicorp/vault-secrets-operator#424]
**Is your feature request related to a problem? Please describe.**
Make it possible to prevent the addition of the `_raw` data to Kubernetes Secrets. This is causing issue with for example the Elastic-Cloud-on-Kubernetes Operator, since it checks the data in the secrets and fails for example if there are other data next to the license json.

Output of the operator logs:
```
"log.level":"ERROR", "message":"fatal exception while booting Elasticsearch", "ecs.version": "1.2.0","service.name":"ES_ECS","event.dataset":"elasticsearch.server","process.thread.name":"main","log.logger":"org.elasticsearch.bootstrap.Elasticsearch","elasticsearch.node.name":"elastic-es-masterdata-2","elasticsearch.cluster.name":"elastic","error.type":"java.lang.IllegalArgumentException","error.message":"unknown secure setting [_raw]
```

**Describe the solution you'd like**
Enable / disable placement of these information using a flag or store it outside of the data section

**Comment by nia-potato** [comment]
same issue here

**Comment by lmanzieri-sbf** [comment]
I have the same problem where the _raw in the k8s secret is breaking the pod tool configuration.

**Comment by sebglon** [comment]
Idem, i have the same problem with elastic operator

**Comment by jmorsecode** [comment]
This is also an issue for any helm-chart options that expect the Secret to include only key:values that are desired. We have a third party chart that takes only the Secret name and creates filenames for each key found. This creates unnecessary _raw files when combined with VSO secrets.

**Comment by apa64** [comment]
The key `_raw` breaks ElasticSearch Enterprise licensing too when running ECK Operator. We can see this in the operator logs:

```
"error":"failed to list enterprise licenses: while parsing license in elastic-system/eck-license: license secret needs to contain exactly one file with any name"
```

**Comment by rgarcia89** [comment]
@benashz is it foreseeable that this improvement will be included in version 0.5.0?

**Comment by benashz** [comment]
> @benashz is it foreseeable that this improvement will be included in version 0.5.0?

@rgarcia89 - yes it should be.
